src/jit/gentree.cpp

```diff
@@ -7817,6 +7817,504 @@ GenTreePtr GenTree::GetChild(unsigned childNum)
     }
 }
 
+GenTreeOperandIterator::GenTreeOperandIterator()
+    : m_node(nullptr)
+    , m_operand(nullptr)
+    , m_argList(nullptr)
+    , m_multiRegArg(nullptr)
+    , m_expandMultiRegArgs(false)
+    , m_state(-1)
+{
+}
+
```

Comment:
Unfortunately it cannot use the existing `GetChild()`: the behavior of the two is different in the case of `GTF_REVERSE_OPS` and nodes that use `GenTreeArgList` nodes internally (calls, phis, and `SIMDIntrinsicInitN`). Even if `GetChild()` produced the same nodes in the same order, replacing this implementation with a loop from 1 to `NumChildren()` over the results of `GetChild()` may be prohibitively expensive in the case of the aforementioned nodes, which would need to repeatedly walk their component lists. I do plan on replacing the existing `NumChildren()`/`GetChild()` over time, but that is orthogonal to this change.